File: frontend/src/shared/types/notification.ts
export type NotificationCategory = 'all' | 'system' | 'chairperson' | 'uk';

export interface NotificationItem {
  id: string;
  category: 'system' | 'chairperson' | 'uk';
  authorName: string;
  authorBadge?: string;
  time: string;
  title: string;
  text: string;
  isUnread: boolean;
  icon?: string;
  avatarText?: string;
}

File: backend/app/models/notification.py
from sqlalchemy import String, Text, ForeignKey, Boolean, Enum
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin
from app.core.constants import NotificationCategory


class Notification(Base, TimestampMixin):
    __tablename__ = "notifications"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    
    category: Mapped[NotificationCategory] = mapped_column(
        Enum(NotificationCategory, name="notification_category_enum"),
        default=NotificationCategory.SYSTEM,
        nullable=False,
    )
    author_name: Mapped[str] = mapped_column(String(255), nullable=False)
    author_badge: Mapped[str | None] = mapped_column(String(50), nullable=True)
    
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    text: Mapped[str] = mapped_column(Text, nullable=False)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False, index=True)
    action_url: Mapped[str | None] = mapped_column(String(255), nullable=True)